[add]#2 Login and sign-up screen

tksx1227 opened the pull request:
ログイン・サインアップ画面、UI面のみ実装完了。

可読性微妙だったら書き方見直すので教えてほしいっす。

tksx1227 committed:
[add]#2 Login and sign-up screen

shwatanap commented on lib/screens/login_signup_screen/login_signup_screen.dart:
```diff
@@ -0,0 +1,16 @@
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Body(),
```
Bodyはwidgetに切らなくていいんじゃないかな？
**_screens.dartを見て大まかにどんなwidgetがあるのかわかる方が望ましいのかなとか思ったり

tksx1227 replied:
OK
ディレクトリ構成の修正と一緒にそこも直してみるわ

shwatanap commented on lib/screens/login_signup_screen/components/input_form.dart:
```diff
@@ -0,0 +1,100 @@
+                          width: double.infinity,
+                        ),
+                        getTextField(false, _hintComfirmPasswordText,
+                            widget.setConfirmPassword),
```
最後にコンマを打って欲しい...

shwatanap commented on lib/screens/login_signup_screen/components/social_icon_buttons.dart:
```diff
@@ -0,0 +1,55 @@
+              color: Colors.black12,
+              blurRadius: 1,
+              spreadRadius: 1.2,
+              offset: Offset(0, 0))
```
ここもコンマを打って欲しい

shwatanap commented on lib/screens/login_signup_screen/components/switch_tab_button.dart:
```diff
@@ -0,0 +1,127 @@
+              color: Colors.black12,
+              blurRadius: 0,
+              spreadRadius: 1.2,
+              offset: Offset(0, 0))
```
ここも

shwatanap commented:
キーボードをカスタムできて、今どっちとも右下のキーが完了(チェクマーク)だと思うんだけど、一番最後のフォーム以外は、nextにして、次のフォームに移れるようにしたほうがいいかなと思う。
参考記事: https://codehero.jp/flutter/52150677/how-to-shift-focus-to-next-textfield-in-flutter

shwatanap commented:
ついでにキーボード外をタップするとキーボードをしまって欲しい
参考記事: https://qiita.com/tetsukick/items/9e07159147308495afc1

tksx1227 committed:
[fix]#2 コメントにて指摘された点の修正

- ディレクトリ構成の修正
- コンマの追加
- キーボード表示中、他の画面に触るとキーボードを閉じるよう修正
- 最後の入力欄より前では、キーボードの完了ボタンが「次へ」になるよう変更
- フォームの位置を固定するよう変更

tksx1227 commented:
一通り修正したお

キーボード出した時に、フォーム動かない方が綺麗だから固定したんだけど、他の端末では被ったりしないかな。

自分で使ってるエミュレータでは全く問題なかったけど。

shwatanap commented on lib/widgets/login_signup_screen/input_form.dart:
```diff
@@ -74,7 +89,8 @@ class _InputFormState extends State<InputForm> {
 
-  Container getTextField(obscureText, hintText, onChanged) {
+  Container getTextField(
+      obscureText, hintText, onChanged, onEditingComplete, isLastElem) {
```
細かくてごめんやけどここもコンマ欲しい

shwatanap commented:
修正確認しました！OKです。
俺もフォームは動かないほうがいいと思うんやけど、ログインボタンがキーボードと被るっていうのはあるかな。
でも動く方が違和感あるから今回の修正文の方が好きです。
<img width="377" alt="スクリーンショット 2021-08-17 19 11 25" src="https://user-images.githubusercontent.com/76590212/129707794-170f8080-7cb9-4cfa-96db-3bf541612c92.png">

tksx1227 committed:
[fix]#2 コンマを追加

tksx1227 commented:
画面の半分より上あたりに入力フォーム出るように調整してるから、このままでも大丈夫そうかな。

tksx1227 committed:
[fix]#2 関数名を一部変更
